tests/rptest/scale_tests/extreme_recovery_test.py

```diff
@@ -0,0 +1,234 @@
+            f"{len(self.topics)} topics, {RecoveryScale.PARTITONS} partitions")
+        t0 = time()
+        # start multiple producers per topic
+        for topic in self.topics:
+            for i in range(self.PRODUCERS_PER_TOPIC):
+                self.logger.info(
+                    f"gen baseline: make producer {i+1} of "
+                    f"{self.PRODUCERS_PER_TOPIC} for {topic.name}")
+                p = FranzGoVerifiableProducer(
+                    self.ctx, self.redpanda, topic.name, self.MSG_SIZE,
+                    self._msg_per_topic // self.PRODUCERS_PER_TOPIC)
+                p.start()
+                producers.append(p)
+
+        # wait for completion
+        for i, p in enumerate(producers):
+            self.redpanda.logger.info(
+                f"gen baseline: wait for producer {i+1}.. ")
+            p.wait(self.expected_transfer_sec)
+        t1 = time()
+        for p in producers:
+            p.free()
+        delta_t = t1 - t0
+        gib = self._total_bytes / 2**30
+        mib = self._total_bytes / 2**20
+        self._status(f"Produced {gib:0.3f} GiB in {delta_t:0.3f}s, "
+                     f"{mib/delta_t:0.3f} MiB/s.")
+
+    def validate_cluster(self, baseline, restored):
+        """Validate all data after recovery."""
+        self.logger.info(
+            f"vallidate_cluster base:restored: {baseline}:{restored}")
+
+        self._validate_partition_last_offset()
+        expected_topics = [
+            topic.name for topic in self.expected_recovered_topics
+        ]
+
+        # consume and validate all data
+        t0 = time()
+        consumers: list[FranzGoVerifiableSeqConsumer] = list()
+        for topic in self.topics:
+            self.logger.info(
+                f'validate_cluster: starting consumer for "{topic}"')
+            c = FranzGoVerifiableSeqConsumer(self.ctx, self.redpanda, topic,
+                                             self.MSG_SIZE)
+            c.start()
+            consumers.append(c)
+
+        total_bytes = 0
+        for c in consumers:
+            self.logger.info(
+                f'validate_cluster: waiting on consumer for "{c._topic}"')
+            c.shutdown()
+            c.wait(self.expected_transfer_sec)
+            total_bytes += c.consumer_status.valid_reads * self.MSG_SIZE
+            assert c.consumer_status.invalid_reads == 0, "No invalid reads expected."
+            assert c.consumer_status.valid_reads == self._msg_per_topic
+
+        delta_t = time() - t0
+        self._status(
+            f'validate_cluster: consumed {total_bytes/2**20:0.1f} MB in {delta_t:0.3f} sec;'
+            f' {(total_bytes/delta_t)/2**20:0.3f} MB/s.')
+
+    @property
+    def second_restart_needed(self):
+        """Return True if the restart is needed afer first validation steps"""
+        # XXX TODO
+        return False
+
+    def after_restart_validation(self):
+        """Check that topic is writable after restart"""
+        # XXX TODO
```

Comment:
Looks like syntax error

Reply:
At least in this commit, it's probably fixed in following commits

Reply:
Thanks. I will add a `pass` so the body won't be empty.